nucleus/common/common-util/src/main/java/com/sun/common/util/logging/GFLogRecord.java

```diff
@@ -84,4 +84,25 @@ public String getThreadName() {
     public void setThreadName(String threadName) {
         this.threadName = threadName;
     }
+
+    /**
+     * CUSTOM-55
+     * in case of an object passed as a parameter, call it's toString() method
+     * to resolve it's values in the current thread, instead of waiting for queues / et
+     * so there is no possibility of state change of the object between thread
+     * @param params
+     * @return parameter array
+     */
+    private Object[] transform(Object[] params) {
+        if (params == null) {
+            return null;
+        }
+        for (int ii = 0; ii < params.length; ++ii) {
+            Object param = params[ii];
+            if (param != null && !param.getClass().equals(String.class)) {
```

Comment:
What is `param.getClass().equals(String.class)` supposed to do?

Reply:
See it’s already a sting. I could use instanceof String instead I guess

Reply:
I think it is not useful at all. String,toString() returns this, so it is NOOP and the condition just obfuscates code. Or am I missing something?

Reply:
No you are not missing anything. You are right it’s superfluous. I will remove that one